backend/routes_rbac.py

```
@@ -8,22 +8,50 @@ def permission_required(permission_name):
-            # Legacy admin users have all permissions
-            if user.is_admin:
                 return f(*args, **kwargs)
 
-            # Check if user has the required permission through roles
-            if not user.has_permission(permission_name):
-                return jsonify({'error': f'Permission {permission_name} required'}), 403
+            except jwt.ExpiredSignatureError:
+                return jsonify({'error': 'Authentication required', 'reason': 'Token expired'}), 401
+            except jwt.InvalidTokenError:
+                return jsonify({'error': 'Authentication required', 'reason': 'Invalid token'}), 401
+            except Exception as e:
```

Ruff (0.11.9)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)